src/main/java/com/bhoruka/bloodbank/dao/entity/BloodComponent.java

```diff
@@ -0,0 +1,113 @@
+package com.bhoruka.bloodbank.dao.entity;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTypeConvertedJson;
+
+import java.util.Date;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import org.springframework.data.annotation.Id;
+
+@NoArgsConstructor
+@DynamoDBTable(tableName = "BloodComponent")
+public class BloodComponent {
+
+    @Id
+    @DynamoDBTypeConvertedJson
+    private BloodComponentId bloodComponentId;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private String bloodGroup;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer wbc;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer pc;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer ffp;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer paedFfp;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer plateletCone;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer cryoPpt;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Integer cfp;
+
+    @DynamoDBAttribute
+    @Getter
+    @Setter
+    private Date date;
+
+    /**
+     * Getter for id.
+     *
+     * @return the blood component id.
+     */
+    @DynamoDBHashKey(attributeName = "id")
+    public String getId() {
```

Comment:
This is because of the foreign key issue? Could you explain here again?
Could we not put the getter and setter annotation at class level?

Reply:
I think this is best explained here https://github.com/derjust/spring-data-dynamodb/issues/24

Reply:
Thank you, this has been very helpful.